services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/system_monitor/_disk_usage.py

```diff
@@ -17,41 +21,177 @@
+            if disk_usage not in usage_to_folder_names:
+                usage_to_folder_names[disk_usage] = set()
+
+            usage_to_folder_names[disk_usage].add(folder_name)
+        return usage_to_folder_names
+
+    async def _publish_disk_usage(self, usage: dict[str, DiskUsage]):
         await publish_disk_usage(
             self.app, user_id=self.user_id, node_id=self.node_id, usage=usage
         )
 
     async def _monitor(self) -> None:
-        disk_usages: list[DiskUsage] = await logged_gather(
-            *[get_usage(monitored_path) for monitored_path in self.monitored_paths]
-        )
+        measured_disk_usage = await self._get_measured_disk_usage()
+
+        normalized_disk_usage = self._get_normalized_disk_usage(measured_disk_usage)
+
+        self._overwrite_with_incoming_disk_usage(normalized_disk_usage)
 
-        usage: dict[Path, DiskUsage] = dict(
-            zip(self.monitored_paths, disk_usages, strict=True)
+        usage_to_folder_names = self._get_grouped_usage_to_folder_names(
+            normalized_disk_usage
         )
 
-        # notify only when changed
+        # compute new version of DiskUsage for FE, only 1 label for each unique disk usage entry
+        usage: dict[str, DiskUsage] = {}
+
+        normalized_paths = self._normalized_monitored_paths
+
+        def _assign_on_match_and_stop(
+            disk_usage: DiskUsage,
+            folder_names: set[str],
+            mount_path_category: MountPathCategory,
+        ) -> bool:
+            # if no match is found returns False so that the next condition can trigger
+            if _have_common_entries(
+                folder_names, normalized_paths[mount_path_category]
+            ):
+                usage[mount_path_category] = disk_usage
+                return True
+            return False
+
+        for disk_usage, folder_names in usage_to_folder_names.items():
+            if not (
+                _assign_on_match_and_stop(
+                    disk_usage, folder_names, MountPathCategory.HOST
+                )
+                or _assign_on_match_and_stop(
+                    disk_usage, folder_names, MountPathCategory.STATES_VOLUMES
+                )
+                or _assign_on_match_and_stop(
+                    disk_usage, folder_names, MountPathCategory.INPUTS_VOLUMES
+                )
+                or _assign_on_match_and_stop(
+                    disk_usage, folder_names, MountPathCategory.OUTPUTS_VOLUMES
+                )
+            ):
+                msg = f"Could not assign {disk_usage=} for {folder_names}"
+                raise RuntimeError(msg)
+
+        detected_items = set(usage.keys())
+        if not detected_items.issubset(_SUPPORTED_ITEMS):
+            msg = (
+                f"Computed {usage=}, has unsupported items {detected_items=}. "
+                f"Currently only  the following are supported: {_SUPPORTED_ITEMS}"
+            )
+            raise RuntimeError(msg)
```

Comment:
This is an exception that can reach your RPC API. My suggestion is:
- create your own custom exceptions inheriting from `OsparcERrorMixin`
- pass an informative error context in the constructor
- handle it before the RPC API or make it part of the API (SEE https://github.com/ITISFoundation/osparc-simcore/pull/6531)

Reply:
No it cannot reach the RPC API, since this task is not called directly by the RPC endpoint (it calls `set_disk_usage_for_path`).
This will only be visible as a traceback in the dy-sidecar logs.

Reply:
@GitHK I still suggest you to use the customize exceptions to get a good troubleshooting log